src/OpenTelemetry/Trace/Export/SimpleSpanProcessor.cs

```diff
@@ -37,14 +45,28 @@ public override void OnStart(Span span)
         /// <inheritdoc />
         public override void OnEnd(Span span)
         {
-            // do not await, just start export
-            this.Exporter.ExportAsync(new[] { span }, CancellationToken.None);
+            try
+            {
+                // do not await, just start export
+                // it can still throw in synchronous part
+                this.Exporter.ExportAsync(new[] { span }, CancellationToken.None);
+            }
+            catch (Exception ex)
+            {
+                OpenTelemetrySdkEventSource.Log.SpanProcessorException(ex);
+            }
         }
 
         /// <inheritdoc />
         public override Task ShutdownAsync(CancellationToken cancellationToken)
         {
-            return this.Exporter.ShutdownAsync(cancellationToken);
+            if (!this.disposed)
+            {
+                this.disposed = true;
+                return this.Exporter.ShutdownAsync(cancellationToken);
```

Comment:
If we don't await here, an exception would be thrown at the caller if they await.
Is it OK to expect that `ShutdownAsync` can throw or would this potentially introduce crashes on app-shutdown?

Reply:
even if we await, an exception will be thrown  (if caller awaits). One part of me tells that whenever you do await with cancellation token, you should expect OperationCancelled and catch. I believe here we try to be explicit and not hide all exceptions from users, so I'd rather throw by default and get more feedback.

If we swallow exceptions, we can never start throwing them. If we throw from the beginning - we can easily stop doing this at any moment.

Reply:
Although It can be a bit frustrating if your instrumentation is throwing an error that bubble up and crash your app, I agree it would be good to surface them first.

And yes, I did mean we'd supress/log them here, essentially hiding it from the user unless they are debugging issues/subscribed to logs.

Reply:
it makes total sense. whatever written in this pr is not final and we'll revisit this question.

You know, we decided to suppress almost all exceptions for `System.Diagnostic.Activity` (which is literally .NET tracing Span). It bites as now people don't even know they do something wrong. I myself make this mistake again and again with Activity APIs :)